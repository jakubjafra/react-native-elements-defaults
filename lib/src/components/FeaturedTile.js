const utils = require('../utils');
const functionalComponentCompose = require('../compose/functionalComponent');

const COMPOSE_NAME = 'FeaturedTile';
const compose = functionalComponentCompose.construct(COMPOSE_NAME);

module.exports = {
    name: COMPOSE_NAME,
    compose,
    override: new utils.ModuleOverride(
        COMPOSE_NAME,
        require('react-native-elements/src/tile/FeaturedTile'),
        compose,
        true,
    ),
};
