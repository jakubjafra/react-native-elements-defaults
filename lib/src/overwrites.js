const Avatar = require('./components/Avatar');
const Badge = require('./components/Badge');
const Button = require('./components/Button');
const ButtonGroup = require('./components/ButtonGroup');
const Card = require('./components/Card');
const CheckBox = require('./components/CheckBox');
const colors = require('./files/colors');
const Divider = require('./components/Divider');
const FeaturedTile = require('./components/FeaturedTile');
const fonts = require('./files/fonts');
const FormInput = require('./components/FormInput');
const FormLabel = require('./components/FormLabel');
const FormValidationMessage = require('./components/FormValidationMessage');
const Header = require('./components/Header');
const Icon = require('./components/Icon');
const List = require('./components/List');
const ListItem = require('./components/ListItem');
const PricingCard = require('./components/PricingCard');
const Rating = require('./components/Rating');
const Search = require('./components/Search');
const Slider = require('./components/Slider');
const SocialIcon = require('./components/SocialIcon');
const Text = require('./components/Text');
const Tile = require('./components/Tile');

module.exports = {
    [Avatar.name]: Avatar,
    [Badge.name]: Badge,
    [Button.name]: Button,
    [ButtonGroup.name]: ButtonGroup,
    [Card.name]: Card,
    [CheckBox.name]: CheckBox,
    [colors.name]: colors,
    [Divider.name]: Divider,
    [FeaturedTile.name]: FeaturedTile,
    [fonts.name]: fonts,
    [FormInput.name]: FormInput,
    [FormLabel.name]: FormLabel,
    [FormValidationMessage.name]: FormValidationMessage,
    [Header.name]: Header,
    [Icon.name]: Icon,
    [List.name]: List,
    [ListItem.name]: ListItem,
    [PricingCard.name]: PricingCard,
    [Rating.name]: Rating,
    [Search.name]: Search,
    [Slider.name]: Slider,
    [SocialIcon.name]: SocialIcon,
    [Text.name]: Text,
    [Tile.name]: Tile,
};
