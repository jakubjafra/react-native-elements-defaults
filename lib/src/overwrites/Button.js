const utils = require('../utils');
const functionalComponentCompose = require('./_funcionalComponentCompose');

const COMPOSE_NAME = 'Button';
const compose = functionalComponentCompose.construct(COMPOSE_NAME);

module.exports = {
    name: COMPOSE_NAME,
    compose,
    override: new utils.ModuleOverride(
        COMPOSE_NAME,
        require('react-native-elements/src/buttons/Button'),
        compose,
        true,
    ),
};
