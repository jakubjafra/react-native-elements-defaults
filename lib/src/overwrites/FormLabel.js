const utils = require('../utils');
const classComponentCompose = require('./_classComponentCompose');

const COMPOSE_NAME = 'FormLabel';
const compose = classComponentCompose.construct(COMPOSE_NAME);

module.exports = {
    name: COMPOSE_NAME,
    compose,
    override: new utils.ModuleOverride(
        COMPOSE_NAME,
        require('react-native-elements/src/form/FormLabel'),
        compose,
        true,
    ),
};
