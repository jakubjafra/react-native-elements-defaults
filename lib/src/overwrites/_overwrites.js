const colors = require('./colors');
const fonts = require('./fonts');
const Text = require('./Text');

module.exports = {
    [colors.name]: colors,
    [fonts.name]: fonts,
    [Text.name]: Text,
};
