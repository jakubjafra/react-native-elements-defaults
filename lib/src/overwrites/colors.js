const utils = require('../utils');
const fileCompose = require('./_fileCompose');

const COMPOSE_NAME = '$colors';
const compose = fileCompose.construct(COMPOSE_NAME);

module.exports = {
    name: COMPOSE_NAME,
    compose,
    override: new utils.ModuleOverride(
        COMPOSE_NAME,
        require('react-native-elements/src/config/colors'),
        compose,
        false,
    ),
};
