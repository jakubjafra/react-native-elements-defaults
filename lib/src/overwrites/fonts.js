const utils = require('../utils');
const fileCompose = require('./_fileCompose');

const COMPOSE_NAME = '$fonts';
const compose = fileCompose.construct(COMPOSE_NAME);

module.exports = {
    name: COMPOSE_NAME,
    compose,
    override: new utils.ModuleOverride(
        COMPOSE_NAME,
        require('react-native-elements/src/config/fonts'),
        compose,
        false,
    ),
};
