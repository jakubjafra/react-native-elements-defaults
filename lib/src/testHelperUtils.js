import 'react-native';
import React from 'react';
import renderer from "react-test-renderer";

const diff = require('jest-diff');

function defaultTests(name,
                      __oldElement__,
                      __element__,
                      basicTheme,
                      createElement = element => React.createElement(element, null, null)) {
    describe('with no theme applied', () => {
        let OldElement, Element;

        beforeAll(() => {
            let defaultStyles = {};

            OldElement = __oldElement__.default;
            Element = __element__.compose(defaultStyles)(OldElement);
        });

        test(`Old${name} and ${name} should be different`, () => {
            expect(OldElement).not.toBe(Element);
        });

        test(`Old${name} renders correctly`, () => {
            const tree = renderer.create(createElement(OldElement)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`${name} renders correctly`, () => {
            const tree = renderer.create(createElement(Element)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`Old${name} and ${name} returns the same thing if no theme is applied`, () => {
            const oldTree = renderer.create(createElement(OldElement)).toJSON();
            const tree = renderer.create(createElement(Element)).toJSON();

            expect(diff(tree, oldTree)).toMatchSnapshot();
        });
    });

    describe('with basic theme applied', () => {
        let OldElement, Element;

        beforeAll(() => {
            let defaultStyles = basicTheme;

            OldElement = __oldElement__.default;
            Element = __element__.compose(defaultStyles)(OldElement);
        });

        test(`Old${name} and ${name} should be different`, () => {
            expect(OldElement).not.toBe(Element);
        });

        test(`Old${name} renders correctly`, () => {
            const tree = renderer.create(createElement(OldElement)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`${name} renders correctly`, () => {
            const tree = renderer.create(createElement(Element)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`Old${name} and ${name} should be different caused by theme applied`, () => {
            const oldTree = renderer.create(createElement(OldElement)).toJSON();
            const tree = renderer.create(createElement(Element)).toJSON();

            expect(oldTree).not.toEqual(tree);
        });
    });
}

module.exports = {
    defaultTests
};
