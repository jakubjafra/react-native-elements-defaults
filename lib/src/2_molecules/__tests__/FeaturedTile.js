import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import * as utils from '../../testHelperUtils';

describe('FeaturedTile', () => {
    utils.defaultTests(
        'FeaturedTile',
        require('react-native-elements/src/tile/FeaturedTile'),
        require('../../overwrites/FeaturedTile'),
        {
            containerStyle: {
                backgroundColor: 'red',
            }
        }
    );
});
