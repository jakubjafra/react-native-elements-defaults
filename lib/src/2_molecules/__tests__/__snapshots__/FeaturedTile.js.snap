// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedTile with basic theme applied FeaturedTile renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "red",
      "height": 600,
      "opacity": 1,
      "width": 750,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "height": 600,
          "justifyContent": "center",
          "resizeMode": "cover",
          "width": 750,
        },
        undefined,
      ]
    }
  >
    <Image
      source={undefined}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "rgba(0,0,0,0.2)",
            "bottom": 0,
            "flex": 1,
            "justifyContent": "center",
            "left": 0,
            "paddingBottom": 40,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 45,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,0)",
                "color": "#ffffff",
                "marginBottom": 15,
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,0)",
                "color": "#ffffff",
                "marginBottom": 15,
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`FeaturedTile with basic theme applied OldFeaturedTile renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 600,
      "opacity": 1,
      "width": 750,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "height": 600,
          "justifyContent": "center",
          "resizeMode": "cover",
          "width": 750,
        },
        undefined,
      ]
    }
  >
    <Image
      source={undefined}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "rgba(0,0,0,0.2)",
            "bottom": 0,
            "flex": 1,
            "justifyContent": "center",
            "left": 0,
            "paddingBottom": 40,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 45,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,0)",
                "color": "#ffffff",
                "marginBottom": 15,
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,0)",
                "color": "#ffffff",
                "marginBottom": 15,
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`FeaturedTile with no theme applied FeaturedTile renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 600,
      "opacity": 1,
      "width": 750,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "height": 600,
          "justifyContent": "center",
          "resizeMode": "cover",
          "width": 750,
        },
        undefined,
      ]
    }
  >
    <Image
      source={undefined}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "rgba(0,0,0,0.2)",
            "bottom": 0,
            "flex": 1,
            "justifyContent": "center",
            "left": 0,
            "paddingBottom": 40,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 45,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,0)",
                "color": "#ffffff",
                "marginBottom": 15,
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,0)",
                "color": "#ffffff",
                "marginBottom": 15,
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`FeaturedTile with no theme applied OldFeaturedTile and FeaturedTile returns the same thing if no theme is applied 1`] = `"[2mCompared values have no visual difference.[22m"`;

exports[`FeaturedTile with no theme applied OldFeaturedTile renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 600,
      "opacity": 1,
      "width": 750,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "height": 600,
          "justifyContent": "center",
          "resizeMode": "cover",
          "width": 750,
        },
        undefined,
      ]
    }
  >
    <Image
      source={undefined}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "rgba(0,0,0,0.2)",
            "bottom": 0,
            "flex": 1,
            "justifyContent": "center",
            "left": 0,
            "paddingBottom": 40,
            "paddingLeft": 25,
            "paddingRight": 25,
            "paddingTop": 45,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            Object {
              "fontSize": 27.5,
            },
            undefined,
            undefined,
            undefined,
            Object {},
            undefined,
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,0)",
                "color": "#ffffff",
                "marginBottom": 15,
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "backgroundColor": "rgba(0,0,0,0)",
                "color": "#ffffff",
                "marginBottom": 15,
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;
